File: api/src/test/java/net/blueshell/api/factory/dto/BlogDTOFactory.java
package net.blueshell.api.factory.dto;

import net.blueshell.api.dto.BlogDTO;
import org.springframework.stereotype.Component;

/**
 * Factory for BlogDTO test instances.
 */
@Component
public class BlogDTOFactory extends BaseDtoFactory<BlogDTO> {

    @Override
    public Class<BlogDTO> targetType() {
        return BlogDTO.class;
    }

    @Override
    public BlogDTO createBasic() {
        BlogDTO dto = new BlogDTO();
        dto.setTitle(unique("Blog Title"));
        dto.setHtml("<p>Test content</p>");
        dto.setPublishedAt(now());
        dto.setUrl("https://example.com/blog/" + nextId());
        return dto;
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/dto/BlogDTOFactory.kt
package net.blueshell.api.factory.dto

import net.blueshell.api.blog.web.dto.BlogDTO
import org.springframework.stereotype.Component

/**
 * Factory for BlogDTO test instances.
 */
@Component
class BlogDTOFactory : BaseDtoFactory<BlogDTO>() {

    override fun targetType(): Class<BlogDTO> = BlogDTO::class.java

    override fun createBasic(): BlogDTO {
        val dto = BlogDTO(
            title = unique("Blog Title"),
            html = "<p>Test content</p>",
            publishedAt = now(),
            url = "https://example.com/blog/${nextId()}"
        )
        return dto
    }
}
